core/src/Kokkos_Macros.hpp

```diff
@@ -541,6 +556,11 @@
 #define KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_CUDA
 #elif defined(KOKKOS_ENABLE_HIP)
 #define KOKKOS_ENABLE_DEFAULT_DEVICE_TYPE_HIP
+#if defined(__HIP__)
+// mark that HIP-clang can use __host__ and __device__
+// as valid overload criteria
+#define KOKKOS_ENABLE_OVERLOAD_HOST_DEVICE
```

Comment:
rename into KOKKOS_IMPL_ENABLE_OVERLOAD_HOST_DEVICE we can NOT expose this to users, since there is no portable way of using this macro.

Reply:
1e612661